de.dlr.sc.virsat.model.edit/src/de/dlr/sc/virsat/model/concept/types/structural/level/HierarchyLevelChecker.java

```diff
@@ -0,0 +1,368 @@
+/*******************************************************************************
+ * Copyright (c) 2008-2019 German Aerospace Center (DLR), Simulation and Software Technology, Germany.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package de.dlr.sc.virsat.model.concept.types.structural.level;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import de.dlr.sc.virsat.model.concept.types.structural.IBeanStructuralElementInstance;
+import de.dlr.sc.virsat.model.concept.types.structural.tree.BeanStructuralTreeTraverser;
+
+/**
+ * <pre>
+ * A class to check elements against a custom tree level structure which are
+ * specified by implementations of {@link IHierarchyLevel}s
+ * 
+ * An example of such a tree structure is a model that consists of a SYSTEM,
+ * SUBSYTEM and EQUIPMENT levels These levels are implemented with the
+ * {@link IHierarchyLevel} interface
+ * 
+ * System MySatellite
+ *  |
+ *  ----Subsystem AOCS
+ *  |	|
+ *  |	----Equipment ReactionWheel1
+ *  |	----Equipment ReactionWheel2
+ *  |
+ *  ----NewElement    <====  This class can check what level this element might be on
+ *  	|
+ *  	----Equipment Thruster
+ * 
+ * </pre>
+ * 
+ * 
+ */
+public class HierarchyLevelChecker {
+
+	protected List<IHierarchyLevel> levels;
+
+	/**
+	 * @param levels
+	 *            A list that specifies the concrete order of tree levels.
+	 * 
+	 *            Levels should not overlap - for any element only one level can
+	 *            have true as result from the
+	 *            {@link IHierarchyLevel#isOnLevel(IBeanStructuralElementInstance)} method.

```

Comment:
Please dont use the links in java doc to exessively. it makes reading javadoc realy hard and the value is very little.

Reply:
The value is that it is a hyperlink. If you read it in the source file, you can F3 or Ctrl+click it, and in the javadoc popup it's also a hyperlink. But sure, we can remove it.